cscs-checks/apps/pytorch/pytorch_horovod_check.py

```diff
@@ -3,82 +3,54 @@
         self.variables = {
             'NCCL_DEBUG': 'INFO',
             'NCCL_IB_HCA': 'ipogif0',
             'NCCL_IB_CUDA_SUPPORT': '1',
-            'OMP_NUM_THREADS': '$SLURM_CPUS_PER_TASK',
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task)
         }
-        hash = 'master'
-        git_url = f'https://raw.githubusercontent.com/horovod/horovod/{hash}/examples/pytorch'  # noqa: E501
-        git_src = 'pytorch_synthetic_benchmark.py'
-        self.prerun_cmds = [f'wget {git_url}/{git_src}']
-
-        if model == 'inception_v3':
-            self.prerun_cmds += [
-                'python3 -m venv --system-site-packages myvenv',
-                'source myvenv/bin/activate',
-                'pip install scipy',
-                'sed -i "s-output = model(data)-output, aux = model(data)-"'
-                f' {git_src}',
-                'sed -i "s-data = torch.randn(args.batch_size, 3, 224, 224)-'
-                f'data = torch.randn(args.batch_size, 3, 299, 299)-"'
-                f' {git_src}'
-            ]
 
-        self.executable = 'python'
-        self.executable_opts = [
-            git_src,
-            f'--model {model}',
-            f'--batch-size {batch_size}',
-            '--num-iters 5',
-            '--num-batches-per-iter 5'
-        ]
-        self.tags = {'production'}
-        self.maintainers = ['RS', 'HM']
-        self.sanity_patterns = sn.all([
-            sn.assert_found(rf'Model: {model}', self.stdout),
-            sn.assert_found(rf'Batch size: {batch_size}', self.stdout)
-        ])
-        self.perf_patterns = {
-            'throughput_per_gpu': sn.extractsingle(
-                r'Img/sec per GPU: (?P<throughput_per_gpu>\S+) \S+',
-                self.stdout, 'throughput_per_gpu', float
-            ),
-            'throughput_per_job': sn.extractsingle(
-                r'Total img/sec on \d+ GPU\(s\): (?P<throughput>\S+) \S+',
-                self.stdout, 'throughput', float
-            ),
-        }
-        ref_per_gpu = 131 if model == 'inception_v3' else 201
-        ref_per_job = ref_per_gpu * mpi_task
-        self.reference = {
-            'dom:gpu': {
-                'throughput_per_gpu': (ref_per_gpu, -0.1, None, 'images/s'),
-                'throughput_per_job': (ref_per_job, -0.1, None, 'images/s'),
-            },
-            'daint:gpu': {
-                'throughput_per_gpu': (ref_per_gpu, -0.1, None, 'images/s'),
-                'throughput_per_job': (ref_per_job, -0.1, None, 'images/s'),
+    # @run_before('performance')
+    # def set_performance(self):
+        ref_per_gpu_sm60 = 131 if self.model == 'inception_v3' else 201
+        ref_total_sm60 = ref_per_gpu_sm60 * self.num_nodes
+        allref = {
+            'sm_60': {
+                'throughput_total': (ref_total_sm60, -0.05, None, 'images/s'),
+                'throughput_iteration': (ref_per_gpu_sm60, -0.05, None,
+                                         'images/s')
```

Comment:
This is not so readable. I would add the `allref` dictionary at the class level, along with separate references for each benchmark. As of the `throughput_total` I would compute it on-the-fly here and add it to the final `reference` dictionary.

Reply:
Here the idea is to create the `allref` dictionary at the class level with only the `'throughput_iteration'` key? and then to define the final reference on the `setup_run` function with something like following?
```
self.reference = {
                '*': {
                    **self.allref['sm_60'][self.model],
                    'throughput_total': (ref_total_sm60, -0.05, None, 'images/s'),
                }
            }
```
where `ref_total_sm60` is computed from the value of `'throughput_iteration'` set on `allref` times `self.num_nodes`?